sys/action.js

```diff
@@ -292,13 +292,31 @@ class ActionService {
                 body: {
                     action: 'query',
                     meta: 'siteinfo|filerepoinfo',
-                    siprop: 'general|namespaces|namespacealiases|specialpagealiases',
+                    siprop: 'general' +
+                        '|namespaces' +
+                        '|namespacealiases' +
+                        '|specialpagealiases' +
+                        '|languagevariants',
                     format: 'json'
                 }
             }, {}, (apiReq, res) => {
                 if (!res || !res.body || !res.body.query || !res.body.query.general) {
                     throw new Error(`SiteInfo is unavailable for ${rp.domain}`);
                 }
+                // Transform from original response format to
+                // {
+                //   'lang' => ['variant1', 'variant2' ... ]
+                // }
+                // for ease of use.
+                const origVariants = res.body.query.languagevariants;
+                const variants = {};
+                if (origVariants) {
+                    Object.keys(origVariants).forEach((lang) => {
+                        variants[lang] = Object.keys(origVariants[lang])
+                        // Filter out non-specific variants like `en`, `zh` etc.
+                        .filter(variant => /-/.test(variant));
```

Comment:
What about multi-lang wikis? Wouldn't `en` be a valid "variant" for commons, e.g.?

Reply:
In case of commons/wikidata it's all pretty different - this is just about conversion, commons actually would return a completely different response. I don't think parsoid even supports that case.